Develop1

Hemull184 opened the pull request.

Hemull184 committed:
add pom.xml

Hemull184 committed:
add BurgerTest

Hemull184 committed:
add BunTest

Hemull184 committed:
add IngredientTest

Hemull184 committed:
remade BunTest

Hemull184 committed:
add IngredientTypeTest

Hemull184 committed:
add Jacoco report and corrected BurgerTest

Hemull184 committed:
add README.md

nikolay-977 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,42 @@
+        this.expectedPrice = expectedPrice;
+    }
+
+    @Parameterized.Parameters
```
Можно улучшить: В параметризованных тестах для аннотации `@Parameterized.Parameters` лучше использовать аргумент name: `@Parameterized.Parameters(name = "Стоимость булочки. Тестовые данные: {0} {1}")`, где {0}, {1} - индексы параметров. С аргументом name наименование теста отобразится в отчете. А использование индексов параметров повысит информативность проверки.

nikolay-977 commented on src/test/java/BunTest.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @Test
+    public void getBunNameTest(){
```
Можно улучшить: для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea используй сочетание клавиш Ctrl+Alt+L. Также можно установить курсор на корневую директорию и нажать клавиш Ctrl+Alt+L, отметить чек-боксы: cleanup code, а также optimize imports(для оптимизации импортов) и rearrange entries(для изменения порядка переменных и методов в соответствии с best practics)

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,76 @@
+        Mockito.when(bun.getName()).thenReturn(buns.get(2).getName());
+        burger.getReceipt();
+        Mockito.verify(bun,Mockito.times(2)).getName();
+        System.out.println(burger.getReceipt().toString());
```
Можно улучшить: если System.out.println() используется для отладки, лучше использовать логирование. https://www.baeldung.com/java-logging-intro

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

Hemull184 committed:
1) В параметризованном тесте добавлен аргумент name
2) Для расстановки пробелов и табуляции использовал сочетание клавиш Ctrl+Alt+L.
3) Подклюил логирование (log4j)
4) Изменил pom.xml  и README.MD
